### DISCUSSION TITLE
Setting for Maximum Number of Note Revisions

### DISCUSSION THREAD

**Discussion by T-Trivium** [TriliumNext/Trilium#4662] [Q&A]
Hello,
Is it possible to set a general setting for a maximum number of [note revisions](https://github.com/zadam/trilium/wiki/Note%20revisions)?

For example, say I wanted to keep the last 25 versions of revisions for a note. I don't necessarily want to delete all revisions (or disable revisions), nor do I want to have an excessive amount of histories kept for a note. Even with updating the note revision snapshot time, there could still be a large amount of revisions for 'long-term' use notes.

Is there a way to limit the number of revisions kept for a note?

Thanks in advance!

**Comment by zadam** [comment]
Hello, no, there isn't such a feature.

**Reply by T-Trivium** [reply]
Hi @zadam,

Thanks for your response and all the hard work you do.

How much of a lift would it be to implement this as a feature in trilium?

**Reply by zadam** [reply]
Sorry, I don't plan to implement many new features, see: https://github.com/zadam/trilium/issues/4620

I try to do mostly bugfixing now.

I've added at least an API method for `BRevision` entity, so that such cleanup could be implemented as a script: https://github.com/zadam/trilium/commit/0ad337c8e806ba84d48d7b97aa46df52d9f236a8

**Reply by T-Trivium** [reply]
Hi @zadam,
I must have missed that announcement. If you ever start to allocate more time to feature improvement, that would be a great one to add. I'll check out that API method as well.

Just want to say a big thank you for all of your efforts! Trilium is fantastic project and I get a lot of use out of it. Completely understand the change in priorities and shift to other things. Your work benefits a lot of people and is very much appreciated. Thank you!
